model/course.go

```diff
@@ -80,33 +93,35 @@ func (c Course) NumStreams() int {
 	return res
 }
 
-//NumUsers returns the number of users enrolled in the course
+// NumUsers returns the number of users enrolled in the course
 func (c Course) NumUsers() int {
 	return len(c.Users)
 }
 
-func (c Course) GetNextLectureDate() time.Time {
-	earliestLecture := time.Now().Add(time.Hour * 24 * 365 * 10) // 10 years from now.
-	for _, s := range c.Streams {
-		if s.Start.Before(earliestLecture) && s.End.After(time.Now()) {
-			earliestLecture = s.Start
-		}
-	}
-	return earliestLecture
-}
-
-func (c Course) GetNextLecture() *Stream {
+// GetNextLecture returns the next lecture of the course
+func (c Course) GetNextLecture() Stream {
+	var earliestLecture Stream
```

Comment:
Not sure if this is related to this change, but on the start page I now see courses without a next lecture on the top: 
![image](https://user-images.githubusercontent.com/44805696/145720823-4241e01b-173d-4a71-9797-27fdc366f164.png)

Reply:
It only works with the hold `GetNextLectureDate` somehow and for me it only happens for the Public Courses. Maybe I can find the bug, but otherwise we can just stick with old method.

Reply:
There seems to be a sorting issue, possibly related with the database retrieval. I stuck with the existing method for now.